Add generic thermostat AC entity config

akking opened the pull request:
## Description:
Add a separate configuration for AC entity so that 'heater' entity is always 'heater'.
The config for 'ac_mode' is used for initial operation mode.

**Related issue (if applicable):** None

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#5360

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

balloob committed:
Version bump to 0.69.0b0

jloutsenhizer committed:
Add missing 'sensor' to ABODE_PLATFORMS (#14313)

This fixes missing light, humidity, temperature sensors from abode component.

balloob committed:
Fix module names for custom components (#14317)

* Fix module names for custom components

* Also set __package__ correctly

* bla

* Remove print

cdce8p committed:
Add debounce to move_cover (#14314)

* Add debounce to move_cover

* Fix spelling mistake

balloob committed:
Revert custom component loading logic (#14327)

* Revert custom component loading logic

* Lint

* Fix tests

* Guard for infinite inserts into sys.path

graffic committed:
fix(hbmqtt): partial packets breaking hbmqtt (#14329)

This issue was fixed in hbmqtt/issues#95 that was released in hbmqtt 0.9.2

balloob committed:
Update netdisco to 1.4.1

balloob committed:
Version bump to 0.69.0b1

balloob committed:
Ignore more loading errors (#14331)

balloob committed:
Version bump to 0.69.0b2

balloob committed:
Bump frontend to 20180509.0

amelchio committed:
Ignore NaN values for influxdb (#14347)

* Ignore NaN values for influxdb

* Catch TypeError

balloob committed:
Version bump to 0.69.0b3

balloob committed:
Remove domain expiry sensor (#14381)

balloob committed:
Version bump to 0.69.0

balloob committed:
Merge pull request #14392 from home-assistant/rc

0.69

tinloaf committed:
Fix waiting for setup that never happens (#14346)

balloob committed:
Version bump to 0.69.1

balloob committed:
Merge pull request #14413 from home-assistant/rc

0.69.1

Commit:
Separate entity configuration for AC

Add a separate configuration for AC entity so that 'heater' entity is always 'heater'.
The config for 'ac_mode' is used for initial operation mode.

Commit:
Separate entity configuration for AC

Add a separate configuration for AC entity so that 'heater' entity is always 'heater'.
The config for 'ac_mode' is used for initial operation mode.

Commit:
Updated test cases and code styple

Update code style.
Updated test cases.

Commit:
Updated test cases and code styple

Update code style.
Updated test cases.

Commit:
Update code style

Commit:
Update code style

scop committed:
Remove extra quotes from docstrings (#14431)

scop committed:
Spelling, grammar etc fixes (#14432)

* Spelling, grammar etc fixes

* s/an api data/data of an api/

scop committed:
Invoke pytest instead of py.test per upstream recommendation, #dropthedot (#14434)

http://blog.pytest.org/2016/whats-new-in-pytest-30/
https://twitter.com/hashtag/dropthedot

cdce8p committed:
Fix fan service description (#14423)

scop committed:
Clean up some Python 3.4 remnants (#14433)

MartinHjelmare committed:
Remove universal wheel setting (#14445)

* Home assistant should not build a universal wheel since we don't
  support Python 2.

fabaff committed:
Upgrade requests_mock to 1.5 (#14444)

fabaff committed:
Upgrade pyota to 2.0.5 (#14442)

* Use constants

* Upgrade pyota to 2.0.5

fabaff committed:
Upgrade python-telegram-bot to 10.1.0 (#14441)

fabaff committed:
Upgrade Sphinx to 1.7.4 (#14439)

fabaff committed:
Upgrade youtube_dl to 2018.05.09 (#14438)

fabaff committed:
Upgrade distro to 1.3.0 (#14436)

PhilRW committed:
Add priority and cycles to LaMetric (#14414)

* Add priority and cycles to LaMetric

Priority can be "info", "warning" (default), or "critical" and
cycles is the number of times the message is displayed. If cycles
is set to 0 we get a persistent notification that has to be dismissed
manually.

* Fix for schema and style

* Fix for style

kellerza committed:
Allow qwikswitch sensors as part of devices (#14454)

Kane610 committed:
Bump requirement to pydeconz v38 (#14452)

fabaff committed:
Add SpaceAPI support (#14204)

* Add SpaceAPI support

* Changes according PR comments

* Add tests

* Remove print

* Minor changes

fabaff committed:
Use ATTR_NAME from const.py (#14450)

rcloran committed:
zha: Fix cluster class check in single-cluster device type (#14303)

zigpy now allows custom devices, which might mean that devices have cluster
objects which are not instances of the default, but may be instances of
sub-classes of the default. This fixes the check for finding single-cluster
device entities to handle sub-classes properly.

akking commented on homeassistant/components/climate/generic_thermostat.py:
```diff
@@ -48,8 +49,9 @@
+    vol.Optional(CONF_HEATER): cv.entity_id,
     vol.Required(CONF_SENSOR): cv.entity_id,
+    vol.Optional(CONF_AC): cv.entity_id,
     vol.Optional(CONF_AC_MODE): cv.boolean,
```
If the set `CONF_AC`, the mode will set to cool. The modes available depend on the if user set either, both or even neither.
[These](https://github.com/home-assistant/home-assistant/pull/14427/files/eee2b64c93fa5a2cf8bfeab55825dea0bea29a99#diff-c3adb9f4cba8997d9148dc1a780f3276R118) code(until line 123) will handle the logic.

akking replied:
Yes, I think that's better way to handle the initial state.

akking commented on homeassistant/components/climate/generic_thermostat.py:
```diff
@@ -48,8 +49,9 @@
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_HEATER): cv.entity_id,
+    vol.Optional(CONF_HEATER): cv.entity_id,
```
Yes, I changed this so user can config AC and heater independently. If user didn't set up any of them, the ` STATE_OFF` will be the only operation mode for this thermostat.

fabaff replied:
Adding a least one switch must be ensured by the configuration validation. Setting the operation mode to `STATE_OFF` doesn't make sense from my point of view as the user will assume that the platform works but it doesn't.

akking replied:
But how can the user assume the platform will work if he/she not providing a switch? 
I agree that we should has a required switch config but shouldn't be just requiring a `heater` conf but user actually want to supply a `ac` switch and then tweak the setting through another config. That seems to be unintuitive and require user to change the switch setting when season changes.

fabaff commented on homeassistant/components/climate/generic_thermostat.py:
```diff
@@ -110,17 +114,25 @@ def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
+            self._operation_list.append(STATE_COOL)
+        if heater_entity_id:
+            self._operation_list.append(STATE_HEAT)
+        self._operation_list.append(STATE_OFF)
```
Initialize the list with `STATE_OFF`.

fabaff committed:
Upgrade pygatt to 3.2.0 (#14447)

schmittx committed:
Add support for direction to fan template (#14371)

* Initial commit

* Update and add tests

Klathmon committed:
Samsung Family hub camera component (#14458)

* add familyhub.py camera

* fix import and REQUIREMENTS

* add to coveragerc

* fix formatting to make houndci-bot happy

* ran scripts/gen_requirements_all.py

* use CONF_IP_ADDRESS

* Revert "ran scripts/gen_requirements_all.py"

This reverts commit 3a38681d8a084e6d4811771ae7a18819477885bc.

* fix library name

* add missing docstrings and enable polling

* Sort imports

MartinHjelmare committed:
Fix key error upon missing node (#14460)

* This is needed after gateway ready message generates an update while
  persistence is off, or while the gateway node hasn't been presented
  yet.

mjg59 committed:
Bump lakeside version (#14471)

This should fix a couple of issues with T1013 bulbs, and also handle
accounts that contain unknown devices.

Commit:
Minor Wink fixes (#14468)

* Updated Wink light supported feature to reflect what features a given light support.

* Fix typo in wink climate

syssi committed:
Don't poll the Samsung Family hub camera (#14473)

dgomes committed:
Optimistic MQTT light (#14401)

* Restores light state, case the light is optimistic

* lint

* hound

* hound

* Added mqtt_json

* hound

* added mqtt_template

* lint

* cleanup

* use ATTR

Commit:
Remove simplepush.io (#14358)

heythisisnate committed:
Add Konnected component with support for discovery, binary sensor and switch (#13670)

* Add Konnected component with support for discovery, binary sensor, and switch

Co-authored-by: Eitan Mosenkis <eitan@mosenkis.net>

* Use more built-in constants from const.py

* Fix switch actuation with low-level trigger

* Quiet logging; Improve schema validation.

* Execute sync request outside of event loop

* Whitespace cleanup

* Cleanup config validation; async device setup

* Update API endpoint for Konnected 2.2.0 changes

* Update async coroutines via @OttoWinter

* Make backwards compatible with Konnected < 2.2.0

* Add constants suggested by @syssi

* Add to CODEOWNERS

* Remove TODO comment

exxamalte committed:
Make Feedreader component more extendable (#14342)

* moved regular updates definition to own method to be able to override behaviour in subclass

* moved filter by max entries to own method to be able to override behaviour in subclass

* event type used when firing events to the bus now based on variable to be able to override behaviour in subclass

* feed id introduced instead of url for storing meta-data about the feed to be able to fetch the same feed from different configs with different filtering rules applied

* keep the status of the last update; continue processing the entries retrieved even if a recoverable error was detected while fetching the feed

* added test cases for feedreader component

* better explanation around breaking change

* fixing lint issues and hound violations

* fixing lint issue

* using assert_called_once_with instead of assert_called_once to make it compatible with python 3.5

gerard33 committed:
BMW code cleanup (#14424)

* Some cleanup for BMW sensors

* Changed dict sort

* Updates based on review and Travis

balloob committed:
Don't add a url to built-in panels (#14456)

* Don't add a url to built-in panels

* Add url_path back

* Lint

* Frontend bump to 20180515.0

* Fix tests

nordlead2005 committed:
Added temperature (apparent) high/low, deprecated max/min (#12233)

amelchio committed:
Improve handling of offline Sonos devices (#14479)

OverloadUT committed:
"unavailable" Media players should be considered off in Universal player (#14466)

The Universal media player inherits the states of the first child player that is not in some sort of "Off" state (including idle.) It was not considering the "unavailable" state to be off. Now it does.

Commit:
Added option to invert aREST pin switch logic for active low relays (#14467)

* Added option to invert aREST pin switch logic for active low relays

* Fixed line lengths

* Changed naming and set optional invert default value.

* Fixed line length

* Removed default from get

scop committed:
Use "Returns" consistently to avoid being treated as section (#14448)

Otherwise, by side effect, results in error D413 by recent pydocstyle.

schmittx committed:
Add HomeKit support for fans (#14351)

n8henrie committed:
Add stdout and stderr to debug output for shell_command (#14465)

balloob committed:
Update frontend to 20180516.0

balloob committed:
Bump frontend to 20180516.1

ChristianKuehnel committed:
New Sensor FinTS (#14334)

mikedm139 committed:
Update hitron_coda.py (#14506)

missed a typo that wasn't caught with testing since I don't have a Rogers router.

heythisisnate committed:
Konnected component follow up (#14491)

* make device_discovered synchronous

* small fixes from code review

* use dispatcher to update sensor state

* update switch state based on response from the device

* interpolate entity_id into dispatcher signal

* cleanup lint

* change coroutine to callback

amelchio committed:
Adjust LimitlessLED properties for running effects (#14481)

thelittlefireman committed:
Bump locationsharinglib to 2.0.2 (#14359)

* Bump locationsharinglib to 2.0.2

* Bump locationsharinglib to 2.0.2

balloob committed:
Update frontend to 20180518.0 (#14510)

* Update frontend to 20180517.0

* Update requirements

* Bump frontend to 20180518.0

exxamalte committed:
Feedreader configurable update interval and max entries (#14487)

dgomes committed:
Fix probability_threshold in binary_sensor.bayesian (#14512) (Closes: #14362)

oblogic7 committed:
Flux bug fix (#14476)

* Simplify conditionals.

* Send white_value on service call.

* Remove extra blank line

* Further simplification of conditionals

* Requested changes

* Do not call getRgb if not needed

* Update log message

fabaff committed:
Show warning if no locations are shared (fixes #14177) (#14511)

cdce8p committed:
Add code to HomeKit lock (#14524)

fabaff committed:
Set pytz to >=2018.04 (#14520)

hanzoh committed:
Add Homematic IP RotaryHandleSensor support (#14522)

* Add Homematic IP RotaryHandleSensor support

HmIP-SRH was in the RotaryHandleSensor class and threw errors that LOWBAT and ERROR could not be found (they are LOW_BAT and SABOTAGE).

* Revert REQUIREMENTS change

cdce8p committed:
Update HAP-python to 2.1.0 (#14528)

OverloadUT committed:
Fix some ISY sensors not getting detected as binary sensors (#14497)

Sensors that were defined via sensor_string were not getting properly identified as binary sensors when they had a uom defining them as binary (the other three methods of detecting binary sensors worked though.)

sdague committed:
Update mychevy to 0.4.0 (#14372)

After 2 months of being offline, the my.chevy website seems to be
working again. Some data structures changed in the mean time. The new
library will handle multiple cars. This involves a breaking change in
slug urls for devices where these now include the car make, model, and
year in them.

Discovery has to be delayed until after the initial site login to get
the car metadata.

balloob committed:
Bump frontend to 20180518.1

fabaff committed:
Set certifi to >=2018.04.16 (#14536)

balloob committed:
Merge branch 'master' into dev

fabaff committed:
Upgrade aiohttp to 3.2.1 (#14517)

* Upgrade aiohttp to 3.2.1

* Upgrade async_timeout to 3.0.0

* Update the order of the requirements

fabaff committed:
Upgrade youtube_dl to 2018.05.18 (#14519)

fabaff committed:
Upgrade keyring to 12.2.1 (#14521)

fabaff committed:
Upgrade restrictedpython to 4.0b4 (#14537)

pavoni committed:
Bump pyvera version (improve stability of poll loop). (#14540)

balloob committed:
Bump frontend to 20180519.0

fabaff committed:
Fix sensor name (fixes #14535) (#14541)

Commit:
Add support to ignore a xiaomi aqara gateway (#14428)

MartinHjelmare committed:
Wait for future mysensors gateway ready (#14398)

* Wait for future mysensors gateway ready

* Add an asyncio future that is done when the gateway reports the
  gateway ready message, I_GATEWAY_READY.
* This will make sure that the gateway is ready before home assistant
  fires the home assistant start event. Automations can now send
  messages to the gateway when home assistant is started.
* Use async timeout to wait max 15 seconds for ready gateway.

* Address comments

Commit:
Pushed to version 0.7.2 of denonavr (#14551)

Oro committed:
Add auto discovery for nanoleaf aurora lights (#14301)

* auto discovery added for nanoleaf aurora lights

* changes requested by review

* visual indentation

* line too long

* hide autocreated config

dmulcahey committed:
zha: Don't poll switch devices (#14560)

dmulcahey committed:
zha: Add metering sensor (#14562)

damarco committed:
zha: Set default binary_sensor state to false (#14553)

cdce8p committed:
Homekit style cleanup (#14556)

* Style cleanup

* Sorted imports
* Harmonized service calls

* Test improvements

* Small update

dmulcahey committed:
Add IlluminanceMeasurementSensor to ZHA (#14563)

* add IlluminanceMeasurementSensor

* address review comment

* Fix whitespace error during merge

rcloran committed:
zha: Bump to zigpy-xbee 0.1.1 (#14566)

balloob committed:
Update frontend to 20180521.0

Oro committed:
fix nanoleaf aurora lights min and max temperature (#14571)

* fixed nanoleaf aurora lights min and max temperature

* review changes

OverloadUT committed:
Fix ISY moisure sensors showing unknown until a leak is detected (#14496)

* Fix ISY leak sensors always showing UNKNOWN until a leak is detected

Added some logic that handles both moisture sensors and door/window sensors

* Handle edge case of leak sensor status update after ISY reboot

If a leak sensor is unknown, due to a recent reboot of the ISY, the status will get updated to dry upon the first heartbeat. This status update is the only way that a leak sensor's status changes without an accompanying Control event, so we need to watch for it.

* Fixes from overnight testing

State was checking the incorrect parameter, and wasn't calling schedule update

* Remove leftover debug log line

* Remove unnecessary pylint instruction

* Remove access of protected property

We can't cast _.status directly to a bool for some unknown reason (possibly with the VarEvents library), but casting to an int then bool does work.

teharris1 committed:
Bump insteonplm version to fix device hanging (#14582)

* Update inteonplm to 0.9.2

* Change to force Travis CI

* Change to force Travis CI

danielperna84 committed:
Update pyhomematic to 0.1.43 (#14583)

* Update __init__.py

* Update requirements_all.txt

tchellomello committed:
Removed attribute current_time from Raincloudy sensors to avoid being triggered by recorder component (#14584)

jaypikay committed:
Lagute LW-12 Wifi LED control (#13307)

* Added platform lw12wifi for Lagute LW-12 Wifi Lights

Supported features:
* RGB colors
* Variable brightness
* 29 effects
* Changing transitions speed for animated effects

* Added lw12wifi to the list of omitted files to test

* Added lw12 module as new requirement for lw12wifi platform

* Added configuration example docstring for platform lw12wifi

* Updating code according to review in PR:

* Removed unused imports: enum, socket.
* Unused and not imported feature SUPPORT_FLASH was removed.
* Unused import lw12 in setup_platform method removed.
* Fixed indention for valuptuous.
* Changed check if effect is None.
* Removed personal debug output.
* Blocking function are not async anymore.

* Further improvements to satisfy PR.

* Unused import asyncio removed.
* Fixed: Return value and docstring no match up for `assumed_state`.

* Check if the set effect is supported, otherwise revert to normal light.

* Added describing missing docstrings to all functions.

* Adopted code to work with HS color setting.

* Syntactical change in comment.

* Removed redefinition of DOMAIN.

* Refactored lw12 controller setup: removed requirement for host and port in LW12Wifi class.

* Rewritten supported feature setup to a more static  expression.

* Removed unused rgb_color property

* Fixed typo in comment for set_light_option

* Changed RGB option validation schema

* Removed instance properties as config options

* Removed optional settings to be more inline with code style.

* Removed unused option from config example

* Removal of unused import

* Added property to disable state polling for this entity.

* Raise an exception if an unknown effect was selected.

* Fixed an issue with the check for known effects.

* As we do not need to set a default, use simple accessing by key.

* Log if an unknown effect was selected.

* Added link to future documentation.

exxamalte committed:
NUT sensor enhancements (#14570) (Fixes #14324)

* removed default value from required parameter; raising PlatformNotReady when connection to nut unavailable; output human-readable state name by default

* removed superfluous sensor name part; showing human-readable form and raw value of current status in more info dialog

* introduced a new virtual sensor type based on the raw status value but used to display a human-readable form of the status

* renamed method

* format string instead of concatenation

* revert the change to the device state attributes - only output the human-readable status without the raw value

fabaff committed:
Add new transmission sensor types (#14530)

SchumyHao committed:
Add Xiaomi Aqara Lock support (#14419)

Commit:
Merge branch 'dev' of https://github.com/akking/home-assistant into dev
